src/org/hitachivantara/ci/build/impl/MavenBuilder.groovy

```diff
@@ -311,7 +315,7 @@ class MavenBuilder extends AbstractBuilder implements IBuilder, Serializable {
             String localSettingsFile = item.settingsFile ?: settingsFile
 
             if (item.containerized) {
-              process("${cmd} -V -s ${localSettingsFile} -Dmaven.repo.local='${localRepoPath}'", steps)
+              process("${cmd} -V -s ${localSettingsFile} -Dmaven.repo.local='${localRepoPath}' -DnodeDownloadRoot='${nodeDownloadRoot}' -DnpmDownloadRoot='${npmDownloadRoot}'", steps)
```

Comment:
Check if `nodeDownloadRoot` and `npmDownloadRoot` are populated and they are then add them to the `cmd` variable. That way we also cover the use case for non [containerised builds](https://github.com/pentaho/pentaho-jenkins-shared-library/blob/master/src/org/hitachivantara/ci/build/impl/MavenBuilder.groovy#L329).

And please add unit tests to cover this new work.